runtests.sh

```diff
@@ -49,7 +49,7 @@ python3 -m coverage && coverage="true"
 
 run_static_tests(){
     SRC_PATHS="bin snapcraft integration_tests snaps_tests"
-    python3 -m flake8 --max-complexity=10 $SRC_PATHS
+    python3 -m flake8 --max-complexity=12 $SRC_PATHS
```

Comment:
Done by adding a specific flake8 comment to disable this check in the function that was failing. 

Rationale is: I can make the complexity "go away" by doing things like a dictionary dispatch but IRL it's replacing a straightforward if/elif chain with something much less obvious.